src/ts/components/subtitletogglebutton.ts

```diff
@@ -57,7 +64,25 @@ export class SubtitleToggleButton extends ToggleButton<ToggleButtonConfig> {
         });
     }
 
+    private initPlayerEvents = () => {
+        this.player.on(this.player.exports.PlayerEvent.SubtitleAdded, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.SubtitleRemoved, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.PeriodSwitch, this.checkSubtitleAvailability);
```

Comment:
I would rather use the `PeriodSwitched` event. This event signals that the content transition has been completed in contrast to `PeriodSwitch` which signals the start of content transition.